src/huggingface_hub/file_download.py

```diff
@@ -328,14 +330,18 @@ def cached_download(
         )
     }
     if isinstance(use_auth_token, str):
-        headers["authorization"] = f"Bearer {use_auth_token}"
+        auth = cast(
+            AuthHeaders, auth_headers(token=use_auth_token, use_basic=use_basic)
+        )
+        headers["authorization"] = auth["authorization"]
```

Comment:
do we really need the cast here? (you've been spoiled by Typescript type inference no? 🤣 )

Reply:
Yes we need the cast, as `auth_header` return type is `Optional` and here it can not be None